config/v1/types_infrastructure.go

```diff
@@ -562,6 +593,33 @@ type BareMetalPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancer defines how the load balancer used by the cluster is configured.
+	// +default={"type": "OpenShiftManagedDefault"}
+	// +kubebuilder:default={"type": "OpenShiftManagedDefault"}
+	// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+	// +optional
+	LoadBalancer *BareMetalPlatformLoadBalancer `json:"loadBalancer,omitempty"`
```

Comment:
This doesn't need to be a pointer, it will be defaulted because it has omitempty, or at least, it shouldn't need to be, unless we are getting complaints from a generator?

Reply:
if I don't set it as pointer, I got a generator error:

```
hack/update-openapi.sh
make[1]: Entering directory '/home/emilien/go/src/github.com/openshift/api/tools'
make[1]: Leaving directory '/home/emilien/go/src/github.com/openshift/api/tools'
I0207 15:00:17.829702 1183293 generator.go:94] Generating openapi definitions
E0207 15:00:25.797196 1183293 openapi.go:321] Error when generating: loadBalancer, LoadBalancer github.com/openshift/api/config/v1.NutanixPlatformLoadBalancer
F0207 15:00:25.797235 1183293 root.go:59] Error running codegen: error running generator openapi: could not generate openapi definitions: error executing openapi generator: Failed executing generator: some packages had errors:
failed to generate default in github.com/openshift/api/config/v1.NutanixPlatformStatus: LoadBalancer: invalid default value (map[string]interface {}{"type":"OpenShiftManagedDefault"}) for non-pointer/non-omitempty. If specified, must be: {}
make: *** [Makefile:82: update-openapi] Error 255
```

Reply:
Ack ok, I think this is to do with how the openapi is used to generate apply configuration, we will have to leave this as is